### DISCUSSION TITLE
Weekly Progress of DP-GEN 20220829

### DISCUSSION THREAD

**Discussion by HuangJiameng** [deepmodeling/dpgen#925] [Show and tell]
# What's new

- We plan to migrate the README to [DP-GEN's document](https://docs.deepmodeling.com/projects/dpgen/en/latest/) soon and are checking and updating the contents of the [manual](https://github.com/deepmodeling/dpgen#dp-gen-manual). Your suggestions are welcome! At the same time, considering that many users are confused about the configuration of parameter files while using DP-GEN, some examples have also been added to [`example`](https://github.com/deepmodeling/dpgen/tree/master/examples), and you can refer to them to prepare parameter files. You can refer to the documentation of dpdispatcher for the parameters of `machine.json`.
Several new volunteers to help us check the documentation together this week. They are researchers from various provinces and cities.
You can follow [DP-GEN's document (devel)](https://docs.deepmodeling.com/projects/dpgen/en/devel/) to see our newly updated documentation. [Contribution is welcome! ](https://docs.deepmodeling.com/projects/dpgen/en/devel/how-to-contribute.html)

- #903
- #910
- #911
Close [#772](https://github.com/deepmodeling/dpgen/issues/772). Close [#776](https://github.com/deepmodeling/dpgen/issues/776).
- #893
You can skip "type_map" or set it to `"auto"`. Then DP-GEN will automatically generate relative atomic masses based on "type_map". If you want to use isotopes, you can specify the relative atomic mass yourself. Note that auto is made up of lowercase letters. 

# Issues

- #889
`init_surf` doesn't support `forward_files` yet.It will be fixed in pr #909
- #502
Might be an element-type-dependent problem of pymatgen.
- #674
Fixed in #894. 
- #600
One can use `para_deg` in `machine.json` to run multiple MD tasks on a GPU workstation.
- #541
The first `type_map` is the one that DP-GEN actually uses. The second one is in `default_training_param` according to the template of DeePMD-kit's training parameters. It will be OK if you skip the second `type_map` in `default_training_param`. Whether you set it or not, the second "type_map" will be assigned the value of the first one.

# Discussion

- #920
If you find that there is a big difference between the total energy of the system and 02.fp/data.000/set.000/energy.npy, while this situation does not appear when comparing the force, here are some suggestions for you.
1. Is your test data in the training set? If not, this may be the case.
2. If the test data exists in the training set, please check whether the units of force and energy are written correctly.
3. Some software will turn the orientation of coordinates. Please check it.
4. You can also scan the potential energy surface with the trained model to see the difference between your machine learning potential and quantification.

- #908 
The value of the fp_task_min from the param.json file only works in the make_train step. In fp step, fp_task_min is '-1' in default. You can set it in model_devi_jobs part of the param.json file. See https://github.com/deepmodeling/dpgen/blob/master/dpgen/generator/run.py#L2465-L2471

- #915
When using plumed in the model_devi process, "atomic relative 1.0" needs to be added in lammps input file. @hu will make a pull request to fix it. Thx to him!

- #922
If you find that after installation, the command "dpgen -h" shows the following error when module Pymatgen has already been installed.
```Python
ModuleNotFoundError: No module named 'pymatgen.analysis.defects'
```
Please check the version of Pymatgen. Pymatgen version v2022.7.19 is the last version with pymatgen.analysis.defects module. An earlier version of Pymatgen is needed.

# Dependency

## Dpdispatcher[#229](https://github.com/deepmodeling/dpdispatcher/pull/229) 
fix TOTP bugs and implement TOTP authentication

**Comment by AnguseZhang** [comment]
Cheers.
